modules/launcher/src/main/java/org/mule/module/launcher/application/DefaultMuleApplication.java

```diff
@@ -38,14 +45,19 @@
 import org.mule.util.ExceptionUtils;
 
 import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.HashMap;
 import java.util.LinkedList;
 import java.util.List;
 import java.util.Map;
+import java.util.Properties;
 
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
 
-public class DefaultMuleApplication implements Application
+public class DefaultMuleApplication extends ConfigurationManagamentAware implements Application
```

Comment:
I don't think that the DefaultMuleApplication (or the DefaultMuleDomain) should be the one with the logic about how to read/write the deployment properties (ConfigurationManagamentAware). I think their factories should receive the properties already loaded into memory and then used and the mechanism to persist/read those should be part of the deployment module.